frontend/src/components/BlockGraph.tsx

```diff
@@ -157,6 +159,7 @@ export function BlockGraph<T>({
               }
               className={classes(css.graphItem, {
                 [css.selected]: index === selected,
+                [css.completed]: completed && completed(item),
```

Comment:
```suggestion
                [css.completed]: completed?.(item),
```

The `?.` operator can also be used for calling nullable functions, or indexing with square brackets `obj?.[key]`